components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -550,9 +550,9 @@ protected String prepareLoginPage(HttpServletRequest request, AuthenticationCont
                 String callbackurl = getCallbackUrl(authenticatorProperties, context);
 
                 String state = getStateParameter(request, context, authenticatorProperties);
-                context.setProperty(OIDCAuthenticatorConstants.AUTHENTICATOR_NAME + STATE_PARAM_SUFFIX, state);
+                context.setProperty(getName() + STATE_PARAM_SUFFIX, state);
                 String nonce = UUID.randomUUID().toString();
-                context.setProperty(OIDC_FEDERATION_NONCE, nonce);
+                context.setProperty(getName() + OIDC_FEDERATION_NONCE, nonce);
```

Comment:
@janakamarasena Is it okay to change the nonce value name by adding the authenticator name as the prefix?

Reply:
+1. Even though this is used in the normal flow and not only dedicated for app native auth it should be okay as this should not be used outside of the authenicator